lib/tasks/uploads.rake

```diff
@@ -107,18 +119,40 @@ def guess_filename(url, raw)
     filename ||= raw[/<a class="attachment" href="(?:https?:)?#{Regexp.escape(url)}">([^<]+)<\/a>/, 1].presence
     filename ||= File.basename(url)
     filename
-  rescue
+  rescue => e
+    puts "Error guessing filename: #{e}"
     nil
   ensure
     f.try(:close!) rescue nil
   end
 end
 
-def migrate_all_from_s3(limit: nil)
-  RailsMultisite::ConnectionManagement.each_connection { migrate_from_s3(limit: limit) }
+def download_s3_upload(upload, max_file_size)
+  download_attempts_remaining = 3
+  file = nil
+  while file.nil? && download_attempts_remaining > 0
+    file = FileHelper.download("http:#{upload.url}", max_file_size: max_file_size, tmp_file_name: "from_s3", follow_redirect: true)
+    # work-around for intermittent download failures
+    # This can be a temporary error if download from S3 fails
+    sleep 1 if file.nil?
+    download_attempts_remaining -= 1
+  end
+  file
+end
+
+def create_local_upload(upload, file, user_id)
+  filename = upload.original_filename
+  origin = upload.origin
+  upload.destroy
+
+  UploadCreator.new(file, filename, origin: origin).create_for(user_id)
+end
```

Comment:
This re-creates uploads, preserving only the actual file and `user_id`, `filename`, `origin` attributes?

I'm fine with creating new upload records (there was even an idea to make them immutable, to have a permanent record of changes) but all original attributes should be preserved.